src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableSortedDictionary_2.Node.cs

```diff
@@ -0,0 +1,878 @@
+            /// </summary>
+            /// <value>
+            /// <c>true</c> if this instance is empty; otherwise, <c>false</c>.
+            /// </value>
+            public bool IsEmpty
+            {
+                get
+                {
+                    Contract.Ensures((_left != null && _right != null) || Contract.Result<bool>());
+                    return _left == null;
+                }
+            }
+
+            /// <summary>
+            /// Gets the left branch of this node.
+            /// </summary>
+            IBinaryTree<KeyValuePair<TKey, TValue>> IBinaryTree<KeyValuePair<TKey, TValue>>.Left
+            {
+                get { return _left; }
+            }
+
+            /// <summary>
+            /// Gets the right branch of this node.
+            /// </summary>
+            IBinaryTree<KeyValuePair<TKey, TValue>> IBinaryTree<KeyValuePair<TKey, TValue>>.Right
+            {
+                get { return _right; }
+            }
+
+            /// <summary>
+            /// Gets the height of the tree beneath this node.
+            /// </summary>
+            public int Height { get { return _height; } }
+
+            /// <summary>
+            /// Gets the left branch of this node.
+            /// </summary>
+            public Node Left { get { return _left; } }
+
+            /// <summary>
+            /// Gets the left branch of this node.
+            /// </summary>
+            IBinaryTree IBinaryTree.Left
+            {
+                get { return _left; }
+            }
+
+            /// <summary>
+            /// Gets the right branch of this node.
+            /// </summary>
+            public Node Right { get { return _right; } }
+
+            /// <summary>
+            /// Gets the right branch of this node.
+            /// </summary>
+            IBinaryTree IBinaryTree.Right
+            {
+                get { return _right; }
+            }
+
+            /// <summary>
+            /// Gets the value represented by the current node.
+            /// </summary>
+            public KeyValuePair<TKey, TValue> Value
+            {
+                get { return new KeyValuePair<TKey, TValue>(_key, _value); }
+            }
+
+            /// <summary>
+            /// Gets the number of elements contained by this node and below.
+            /// </summary>
+            int IBinaryTree.Count
+            {
+                get { throw new NotSupportedException(); }
+            }
+
+            /// <summary>
+            /// Gets the keys.
+            /// </summary>
+            internal IEnumerable<TKey> Keys
+            {
+                get { return Linq.Enumerable.Select(this, p => p.Key); }
+            }
+
+            /// <summary>
+            /// Gets the values.
+            /// </summary>
+            internal IEnumerable<TValue> Values
+            {
+                get { return Linq.Enumerable.Select(this, p => p.Value); }
```

Comment:
Besides moving files out, I made one tiny code change. Before, this was

```cs
this.Select(p => p.Value);
```

I changed it and removed the System.Linq using statement to lower the changes of one of the functions here accidentally calling Linq.